File: Semester 4/Programming and Projection Enironments/mpp-proiect-csharp/Proiect_MPP/Proiect_MPP/domain/Flight.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Proiect_MPP.domain
{
    public partial class Flight : Entity<int>
    {
        private int freeSeats;
        private int destinationAirport;
        private int departureAirport;
        private DateTime departureDate;
        private DateTime departureTime;

        // Class Constructors //

        public Flight() : base(0)
        {
            this.freeSeats = 0;
            this.destinationAirport = 0;
            this.departureAirport = 0;
            this.departureDate = DateTime.MinValue;
            this.departureTime = DateTime.MinValue;
        }

        public Flight(int id, int freeSeats, int destinationAirport, int departureAirport, DateTime departureDate, DateTime departureTime) : base(id)
        {
            this.freeSeats = freeSeats;
            this.destinationAirport = destinationAirport;
            this.departureAirport = departureAirport;
            this.departureDate = departureDate;
            this.departureTime = departureTime;
        }

        // Getters & Setters //

        public int FreeSeats
        {
            get { return this.freeSeats; }
            set { this.freeSeats = value; }
        }

        public int DestinationAirport
        {
            get { return this.destinationAirport; }
            set { this.destinationAirport = value; }
        }

        public int DepartureAirport
        {
            get { return this.departureAirport; }
            set { this.departureAirport = value; }
        }

        public DateTime DepartureDate
        {
            get { return this.departureDate; }
            set { this.departureDate = value; }
        }

        public DateTime DepartureTime
        {
            get { return this.departureTime; }
            set { this.departureTime = value; }
        }

        // ToString & other functions //

        public override string ToString()
        {
            return "Flight{" +
                "id=" + base.ID +
                ", freeSeats=" + freeSeats +
                ", destination='" + destinationAirport + '\'' +
                ", airportName='" + departureAirport + '\'' +
                ", departureDate=" + departureDate +
                ", departureTime=" + departureTime +
                '}';
        }
    }
}

File: Semester 4/Programming and Projection Enironments/mpp-proiect-java/Proiect_MPP/Thrift/src/main/java/root/proiect_mpp/domain/Flight.java
package root.proiect_mpp.domain;

import java.time.LocalDate;
import java.time.LocalTime;

public class Flight implements Entity<Integer> {
    private int id;
    private int freeSeats;
    private int destinationAirport;
    private int departureAirport;
    private LocalDate departureDate;
    private LocalTime departureTime;

    // Class Constructors //

    public Flight() {
        this.id = 0;
        this.freeSeats = 0;
        this.destinationAirport = 0;
        this.departureAirport = 0;
        this.departureDate = LocalDate.now();
        this.departureTime = LocalTime.now();
    }

    public Flight(int freeSeats, int destinationAirport, int departureAirport, LocalDate departureDate, LocalTime departureTime) {
        this.id = 0;
        this.freeSeats = freeSeats;
        this.destinationAirport = destinationAirport;
        this.departureAirport = departureAirport;
        this.departureDate = departureDate;
        this.departureTime = departureTime;
    }

    public Flight(int id, int freeSeats, int destinationAirport, int departureAirport, LocalDate departureDate, LocalTime departureTime) {
        this.id = id;
        this.freeSeats = freeSeats;
        this.destinationAirport = destinationAirport;
        this.departureAirport = departureAirport;
        this.departureDate = departureDate;
        this.departureTime = departureTime;
    }

    // Getters & Setters //

    @Override
    public Integer getId() {
        return id;
    }

    @Override
    public void setId(Integer id) {
        this.id = id;
    }

    public int getFreeSeats() {
        return freeSeats;
    }

    public void setFreeSeats(int freeSeats) {
        this.freeSeats = freeSeats;
    }

    public int getDestinationAirport() {
        return destinationAirport;
    }

    public void setDestinationAirport(int destinationAirport) {
        this.destinationAirport = destinationAirport;
    }

    public int getDepartureAirport() {
        return departureAirport;
    }

    public void setDepartureAirport(int departureAirport) {
        this.departureAirport = departureAirport;
    }

    public LocalDate getDepartureDate() {
        return departureDate;
    }

    public void setDepartureDate(LocalDate departureDate) {
        this.departureDate = departureDate;
    }

    public LocalTime getDepartureTime() {
        return departureTime;
    }

    public void setDepartureTime(LocalTime departureTime) {
        this.departureTime = departureTime;
    }

    // toString & other functions //

    @Override
    public String toString() {
        return "Flight{" +
                "id=" + id +
                ", freeSeats=" + freeSeats +
                ", destination airport id='" + destinationAirport + '\'' +
                ", departure airport id='" + departureAirport + '\'' +
                ", departureDate=" + departureDate +
                ", departureTime=" + departureTime +
                '}';
    }
}
